File: POO/JAVA/Computador.java
public class Computador {

    //Atributos (con modificadores de acceso y diferentes tipos de datos)
    private String marca;          // texto
    protected int memoriaRAM;      // numérico entero
    public boolean encendido;      // booleano
    private String infoJSON;       // texto (JSON simulado)
    private byte[] datosSistema;   // tipo binario

    //Constructor
    public Computador(String marca, int memoriaRAM, boolean encendido) {
        this.marca = marca;
        this.memoriaRAM = memoriaRAM;
        this.encendido = encendido;
        this.infoJSON = String.format("{\"marca\":\"%s\",\"ram\":%d}", marca, memoriaRAM);
        this.datosSistema = ("SO-" + marca).getBytes();
    }

    //Método SIN parámetros y SIN retorno
    public void encender() {
        encendido = true;
        System.out.println("El computador se ha encendido.");
    }

    //Método CON parámetros y SIN retorno
    public void actualizarRAM(int nuevaRAM) {
        memoriaRAM = nuevaRAM;
        System.out.println("La RAM ha sido actualizada a " + nuevaRAM + " GB.");
    }

    //Método SIN parámetros y CON retorno
    public int obtenerRAM() {
        return memoriaRAM;
    }

    //Método CON parámetros y CON retorno
    public String compararMarca(String otraMarca) {
        if (marca.equals(otraMarca)) {
            return "Es la misma marca.";
        } else {
            return "Es una marca diferente.";
        }
    }

    //Encapsulación (get y set)
    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    //Función (abstracción)
    public String mostrarInfo() {
        return "Computador: " + marca + ", RAM: " + memoriaRAM + " GB, Encendido: " + encendido;
    }
}

File: POO/GO/Computador.go
package main

import (
	"encoding/json"
	"fmt"
)

// Estructura Computador (similar a clase)
type Computador struct {
	marca          string
	memoriaRAM     int
	Encendido      bool
	infoJSON       string
	datosSistema   []byte
}

// Constructor
func NuevoComputador(marca string, memoriaRAM int, encendido bool) Computador {
	return Computador{
		marca:        marca,
		memoriaRAM:   memoriaRAM,
		Encendido:    encendido,
		infoJSON:     fmt.Sprintf(`{"marca":"%s","ram":%d}`, marca, memoriaRAM),
		datosSistema: []byte("SO-" + marca),
	}
}

// Método SIN parámetros y SIN retorno
func (c *Computador) Encender() {
	c.Encendido = true
	fmt.Println("El computador se ha encendido.")
}

// Método CON parámetros y SIN retorno
func (c *Computador) ActualizarRAM(nuevaRAM int) {
	c.memoriaRAM = nuevaRAM
	fmt.Println("RAM actualizada a", nuevaRAM, "GB.")
}

// Método SIN parámetros y CON retorno
func (c Computador) ObtenerRAM() int {
	return c.memoriaRAM
}

// Método CON parámetros y CON retorno
func (c Computador) CompararMarca(otra string) string {
	if c.marca == otra {
		return "Es la misma marca."
	}
	return "Es una marca diferente."
}

// Get y Set
func (c *Computador) GetMarca() string {
	return c.marca
}

func (c *Computador) SetMarca(m string) {
	c.marca = m
}

// Abstracción
func (c Computador) MostrarInfo() string {
	return fmt.Sprintf("Computador: %s, RAM: %d GB, Encendido: %t", c.marca, c.memoriaRAM, c.Encendido)
}
